cibuildwheel/docker_container.py

```diff
@@ -57,39 +80,49 @@ def __enter__(self) -> "DockerContainer":
             network_args = ["--network=host"]
 
         shell_args = ["linux32", "/bin/bash"] if self.simulate_32_bit else ["/bin/bash"]
+
         subprocess.run(
             [
-                "docker",
+                self.container_engine,
                 "create",
                 "--env=CIBUILDWHEEL",
                 f"--name={self.name}",
                 "--interactive",
-                "--volume=/:/host",  # ignored on CircleCI
                 *network_args,
-                *cwd_args,
+                # Z-flags is for SELinux
+                "--volume=/:/host:Z",  # ignored on CircleCI
                 self.docker_image,
                 *shell_args,
             ],
+            env=self.env,
             check=True,
         )
+
         self.process = subprocess.Popen(
             [
-                "docker",
+                self.container_engine,
                 "start",
                 "--attach",
                 "--interactive",
                 self.name,
             ],
             stdin=subprocess.PIPE,
             stdout=subprocess.PIPE,
+            env=self.env,
         )
 
         assert self.process.stdin and self.process.stdout
         self.bash_stdin = self.process.stdin
         self.bash_stdout = self.process.stdout
 
         # run a noop command to block until the container is responding
-        self.call(["/bin/true"])
+        self.call(["/bin/true"], cwd="")
+
+        if self.cwd:
+            # Although `docker create -w` does create the working dir if it
+            # does not exist, podman does not. There does not seem to be a way
+            # to setup a workdir for a container running in podman.
+            self.call(["mkdir", "-p", str(self.cwd)], cwd="")
```

Comment:
Thinking about alternative approaches here... is the problem that the directory doesn't exist or that podman doesn't support `-w`? Perhaps we could do a `podman exec mkdir -p {self.cwd}` before attaching to the container?

Reply:
I think it is the second one. It does look like it has `-w`, but I think the create step itself fails if the "image" doesn't have the directory already in it, which we can't do anything about (AFAIK). I can try it if you think it is cleaner, but the difference between the methods seems like a wash to me.

Reply:
Ah, interesting, so it's the create step that fails. I guess that makes sense given that's where the `-w` is applied. So the `exec` approach wouldn't work then.

Reply:
To close the loop here, I have a MWE. It is not the create step that fails, but it seems that the start step fails. I'm not sure if you can apply an exec before the container is started, it gave me an error when I tried. However, it worked fine under docker.

```bash
# Create a podman container with "-w"
podman create --name=foobar-podman-test --interactive -w "/createme-dir" python:3.7 /bin/bash

# Running the exec here will fail because the container is not started
podman exec -i foobar-podman-test mkdir -p "/createme-dir"

# But starting the container raises an error when create was used with "-w"
podman start foobar-podman-test

# Cleanup
podman rm --force -v foobar-podman-test
```

Here is similar logic working with docker: 

```bash
# Demo where this sequence works with docker
docker create --name=foobar-docker-test --interactive -w "/createme-dir"  python:3.7 /bin/bash
docker start foobar-docker-test
docker exec -i foobar-docker-test mkdir -p "/createme-dir"
docker rm --force -v foobar-docker-test
```

I've documented this MWE in my misc repo: https://github.com/Erotemic/misc/blob/main/tests/containers/test_container_create_with_cwd.sh so I don't lose it.